### PR TITLE
Create fee module class - Closes #6692

### LINKED ISSUE #6692: Create module class, constants, schemas for fee module
### Description

-  Create fee module class
-  Define `addDependencies` class method 
-  Define `token api interface` 
-  Define constants (constants.ts) and schema (schema.ts)
-  Define configurations which are received from the `init` hook 
-  Define hooks `verifyTransaction` and `beforeTransactionExecution`

Follow: https://research.lisk.com/t/introduce-fee-module/318

### Acceptance Criteria
- All files and classes exist in the fee module folder

### Additional Information

Related issue: #6691

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #6692

### How was it solved?

🌱 81db6898d2b8aa0d7964d921489e1c4976751dae

-  Create fee module class
-  Add `addDependencies` method to interact with token module
-  Add configuration options
-  Add `verifyTransaction` and `beforeTransactionExecution` hooks

🌱 16474ee48e2609cc0f88c300e20d04e21464b294

- Add schemas for configuration inputs

### How was it tested?

To be added later.

### CODE REVIEW

File: framework/src/modules/fee/module.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	TransactionExecuteContext,
+	TransactionVerifyContext,
+	VerificationResult,
+} from '../../node/state_machine';
+import { FeeAPI } from './api';
+import { FeeEndpoint } from './endpoint';
+import { configSchema } from './schemas';
+
+export class FeeModule extends BaseModule {
+	public id = 11;
```
Comment (shuse2) [inline framework/src/modules/fee/module.ts:27]: this should be `MODULE_ID_FEE` defined in the constants

**Review by shuse2 — CHANGES_REQUESTED** [review]
one minor comment but LGTM

**Review by shuse2 — APPROVED** [review]

File: framework/src/modules/fee/types.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		localID: number;
+	};
+}
+
+export interface TokenAPI {
+	transfer: (
+		apiContext: APIContext,
+		senderAddress: Buffer,
+		generatorAddress: Buffer,
+		id: { chainID: number; localID: number },
```
Comment (ManuGowda) [inline framework/src/modules/fee/types.ts:35]:
```suggestion
		feeTokenID: FeeTokenID,
```

File: framework/src/modules/fee/types.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		amount: bigint,
+	) => Promise<void>;
+	burn: (
+		apiContext: APIContext,
+		senderAddress: Buffer,
+		id: { chainID: number; localID: number },
```
Comment (ManuGowda) [inline framework/src/modules/fee/types.ts:41]:
```suggestion
		feeTokenID: FeeTokenID,
```

File: framework/src/modules/fee/types.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { APIContext } from '../../node/state_machine';
+
+export interface BaseFee {
+	moduleID: number;
+	commandID: number;
+	baseFee: bigint;
+}
+
+export interface ModuleConfig {
+	feeTokenID: {
```
Comment (ManuGowda) [inline framework/src/modules/fee/types.ts:24]: you can extract this to `FeeTokenID` interface
```
interface FeeTokenID {
  chainID: number;
  localID: number;
}
```

then

`feeTokenID: FeeTokenID`

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**